airbyte-webapp/src/components/connectorBuilder/Builder/InputsView.tsx

```diff
@@ -0,0 +1,303 @@
+              </li>
+            ))}
+          </ol>
+        </Card>
+      )}
+      <Button
+        className={styles.addInputButton}
+        onClick={() => {
+          setInputInEditing(newInputInEditing());
+        }}
+        icon={<FontAwesomeIcon icon={faPlus} />}
+        iconPosition="left"
+        variant="secondary"
+      >
+        <FormattedMessage id="connectorBuilder.addInputButton" />
+      </Button>
+      {inputInEditing && (
+        <Formik
+          initialValues={inputInEditing}
+          validationSchema={inputInEditValidation}
+          onSubmit={(values: InputInEditing) => {
+            const newInput = inputInEditingToFormInput(values);
+            helpers.setValue(
+              inputInEditing.isNew
+                ? [...inputs.value, newInput]
+                : inputs.value.map((input) => (input.key === inputInEditing.key ? newInput : input))
+            );
+            setInputInEditing(undefined);
+          }}
+        >
+          <>
+            <FormikPatch />
+            <InputModal
+              inputInEditing={inputInEditing}
+              onDelete={() => {
+                helpers.setValue(inputs.value.filter((input) => input.key !== inputInEditing.key));
+                setInputInEditing(undefined);
+              }}
+              onClose={() => {
+                setInputInEditing(undefined);
+              }}
+            />
+          </>
+        </Formik>
+      )}
+    </BuilderConfigView>
+  );
+};
+const InputModal = ({
+  inputInEditing,
+  onClose,
+  onDelete,
+}: {
+  inputInEditing: InputInEditing;
+  onDelete: () => void;
+  onClose: () => void;
+}) => {
+  const { isValid, values, setFieldValue } = useFormikContext<InputInEditing>();
+  const { formatMessage } = useIntl();
+  const [title, titleMeta] = useField<string | undefined>("definition.title");
+  useEffect(() => {
+    if (titleMeta.touched) {
+      setFieldValue("key", sluggify(title.value || ""));
+    }
+  }, [setFieldValue, title.value, titleMeta.touched]);
+
+  return (
+    <Modal
+      size="sm"
+      title={
+        <FormattedMessage
+          id={inputInEditing.isNew ? "connectorBuilder.inputModal.newTitle" : "connectorBuilder.inputModal.editTitle"}
+        />
+      }
+      onClose={onClose}
+    >
+      <Form>
+        <ModalBody className={styles.inputForm}>
+          <BuilderField
+            path="definition.title"
+            type="string"
+            label={formatMessage({ id: "connectorBuilder.inputModal.inputName" })}
+            tooltip={formatMessage({ id: "connectorBuilder.inputModal.inputNameTooltip" })}
+          />
+          <BuilderField
+            path="key"
+            type="string"
+            readOnly
+            label={formatMessage({ id: "connectorBuilder.inputModal.fieldId" })}
+            tooltip={formatMessage(
+              { id: "connectorBuilder.inputModal.fieldIdTooltip" },
+              {
+                syntaxExample: "{{my_input}}",
```

Comment:
I tried interpolating a test input (`pokemon_name`) that I added by setting the Path URL of a stream to `/pokemon/{{pokemon_name}}` , but it seems like the value I type into the Test Input menu is not being used.

Originally I thought this was because doing `{{field_id}}` was not actually the right way to interpolate it, as all of the examples in the low-code docs do `{{config['field_id']}}` to interpolate user input values. But even when I tried that, it still didn't work. (Though, this `syntaxExample` should still be updated to say something like `{{config['my_input']}}`, and we may want to file a separate issue for allowing users to just enter `{{my_input}}` instead -- thoughts?)

I just double checked and this is working on the `feature/connector-builder` branch: Open that branch, switch to the YAML view, and paste in this YAML:
```
version: "0.1.0"

streams:
  - type: DeclarativeStream
    $options:
      name: "pokemon"
      primary_key: "id"
      path: "/pokemon/{{config['pokemon_name']}}"
    primary_key: "id"
    retriever:
      requester:
        url_base: "https://pokeapi.co/api/v2"
        http_method: "GET"
      record_selector:
        extractor:
            field_pointer: []

check:
  stream_names:
    - "customers"

spec:
  documentation_url: https://docsurl.com
  connectionSpecification:
    title: Poke API Spec
    type: object
    required:
      - pokemon_name
    additionalProperties: true
    properties:
      pokemon_name:
        type: string
        description: Pokemon name
```

Then open the test inputs menu and enter `ditto` into the pokemon_name field. You can see that the URL is correctly pulling in the `ditto` value, and clicking Test properly fetches data about ditto.

But if you try the same thing on this branch, `ditto` isn't used in the URL. So this seems to be a regression specific to this PR.

Reply:
Yeah, I noticed this as well and fixed it in the authentication PR: https://github.com/airbytehq/airbyte/pull/20645/files#diff-05522b7f5df1df909f26423e40703073e6ab13f90f10ceeef5ca00a65561b351R88

Pulled it into this branch as well.